### ISSUE TITLE
autoit.c OOB read DoS

### ISSUE THREAD

**Issue by felixgr** [vrtadmin/clamav-devel#39]
In autoit.c:585 a new buffer is alloced. In the repro example UNP.csize is 2. However, a subsequent readint32 on line 598 reads out-of-bounds of the allocated buffer.

Repro.example.bzip2.base64:

```
QlpoOTFBWSZTWcOp+lAAAZL////34cXzZumP+3+7t/////////x6+nf1dV3c2UfNafvP0ANW3t3d
27cpbOAZKTRJ6ag/Sn6kaZNqGmIeiAMI9IHogDT1GmmT0mmmjQybUPTSZogNAYRiAaZNGjTRhBpo
zTQTwoNEpqaGRoI2pgmRgepN6kxomEYTIyYmCYAARgJkYQyMRkYA0TIZDRowmGgBMmQwEiVKeoDI
DQMIAAABpk0AA0NADIAaGgGIMQAaDQAAAAABkAaAIBkGgAZNMQaaA0ZMmCZGjQyGgANGQaMIZDJk
YIaADQaYgMEaA0GgBoABIqQmhNABk9QGgNGgD1NNBkA0ANAAAD1AHqBiDRoaAAANAGg0APUBo2o0
MVpBWyOqVBTFU6CS0GsPy56G1cuFehIBqaCQYBgwCJigM8hJFjYrC0H7iAl7pACJO/jV6qKR1l5h
aJ0FwX+Jaymo5DvoVorjXnQBYimOrAdoS+KlEvXGlqQKg1RVAlDRcYviv0gkBJC0qZJIMDGuKoeG
M5AckqoA+zLVULCkJJV6YTzPR0I3bRiEZUeWG2EGyGkmvzIky0mJeZn/svafMZd5BidEeEuoIsUR
U4rREFbEzB/zSXjoiqgJMKkxUBcj7pchQ6YxQ3MHNiVpqnFID00IIHFTlRqWMfkp+40K2/OoVVZC
Mq7KQbA2AYXpogaAtzSJtoNewW7YorQEvO2UidqeeU4wQVjbQNobSTYBZ56F9FR0ioKG1SJmUVFQ
ltLw9LgOnNnage+wPcuDzz3sAZEAzRX1D+Uj5gzhWOlaM3TGFc4TGAZcWxpxDFPc3+GPgoWKesJk
pitXThW6eb4UoNYdAgnLUTz5dgrIwYMwZmDGhsx42EIgYUTGwcENtDG0MbGMbGSTHVAkg1NDGxc8
hBIYA2JBUBNBXfcprb6ZWRCjCnjq+x/UmguIMG0JsG0NpsGMHs4hgNttgdjt9oBKxNCJIENKACRz
KeNMei71NeciznSws6I50/ONCRKaQiGgKVibBNgeVwS4AFWtCcKqDMohBQ0eEnshYBwisCTMNUlJ
Cpu7XK0IV8YDGNpsE2CwrCGkT7IaTHKYq2YmjZtMA+MBUrKTqAGD4QS0jCCkQ1kpUlpgL+x5qJek
/thGvIyYFiGthqGTNnAo4w91EDUpXJSw3vyWMVlEJox4Vk4tjQECiKH+6AhLpKCyn7FRK1om43kv
mfCWFsIob9h/GiUwpUPtsc+GpnMLMhIoV6/cjwtPOjgTFP/GrBGebVoO6r+UCsF6JMs8YQ5UDPCK
EBvUMEZaZ3AKoFKh2eyYXSzENUHMbQRaN8wsjA5pU/wuFABU5YpjKgLu2RNBeILNgInLKgDkoPSZ
8CE2tGJyahpNP3B8cYbyje/OLbsrJIcOJemGIr557AzLeL2Xsm8c6PaifRFmyjVAt1VC6eP/+LuS
KcKEhh1P0oA=
```

**Comment by felixgr** [comment]
A similar problem occurs on line 277.

**Comment by steve-morgan** [comment]
Tracking with https://bugzilla.clamav.net/show_bug.cgi?id=11446.

**Comment by felixgr** [comment]
Was there any progress? Can you add groebert at google.com to the ACL of the bugzilla bug?

**Comment by steve-morgan** [comment]
Fix shipped with 0.99.1.
